subsys/bluetooth/host/l2cap.c

```diff
@@ -913,18 +913,18 @@ static void le_disconn_req(struct bt_l2cap *l2cap, u8_t ident,
 	struct bt_l2cap_le_chan *chan;
 	struct bt_l2cap_disconn_req *req = (void *)buf->data;
 	struct bt_l2cap_disconn_rsp *rsp;
-	u16_t scid;
+	u16_t dcid;
 
 	if (buf->len < sizeof(*req)) {
 		BT_ERR("Too small LE conn req packet size");
 		return;
 	}
 
-	scid = sys_le16_to_cpu(req->scid);
+	dcid = sys_le16_to_cpu(req->dcid);
 
-	BT_DBG("scid 0x%04x dcid 0x%04x", scid, sys_le16_to_cpu(req->dcid));
+	BT_DBG("dcid 0x%04x scid 0x%04x", dcid, sys_le16_to_cpu(req->scid));
 
-	chan = l2cap_remove_tx_cid(conn, scid);
+	chan = l2cap_remove_tx_cid(conn, dcid);
```

Comment:
I think this particular bug may be because of the function naming. Why is it called `l2cap_remove_tx_cid()` if it's operating on `BT_L2CAP_LE_CHAN(chan)->rx.cid`?

Also, do we have a bug in `le_disconn_rsp()` since it also uses `l2cap_remove_tx_cid()`? The L2CAP spec says this about the disconnect response: "Destination CID - DCID (2 octets)
This field identifies the channel endpoint on the device sending the response." However in our current implementation the association is made between the DCID and rx.cid.

Reply:
I fixed the le_disconn_rsp and renamed remove_tx_cid to remove_rx_cid since we are always removing by the local endpoint not the remote one.

I guess we didn't capture this before becuase the cid were the same, but the Linux side does seems to be right and with this changes we have similar logic:

Disconnect request:
https://git.kernel.org/pub/scm/linux/kernel/git/bluetooth/bluetooth-next.git/tree/net/bluetooth/l2cap_core.c#n4320

Disconnect response:
https://git.kernel.org/pub/scm/linux/kernel/git/bluetooth/bluetooth-next.git/tree/net/bluetooth/l2cap_core.c#n4366